packages/contracts/contracts/deploy/PayableDAppSystem.sol

```diff
@@ -16,9 +17,25 @@ contract PayableDAppSystem is IPayableDAppSystem {
     ENS public immutable ens;
     ICartesiDAppFactory public immutable factory;
 
-    constructor(ENS _ens, ICartesiDAppFactory _factory) {
+    uint256 public constant TOTAL_SHARES = 10000;
```

Comment:
Why not make `TOTAL_SHARES` a constructor argument, like `systemShares`?
Say you want to have one third of the shares for the factory.
With `TOTAL_SHARES` being set to `10000`, you're prone to rounding errors ($.3333 \ne \frac{1}{3}$).